wagtail/search/tests/test_postgres_backend.py

```diff
@@ -208,3 +208,99 @@ def test_search_language_phrase_text(self):
 
         results = self.backend.search(Phrase("Nu zijn beter"), models.Book)
         self.assertEqual(list(results), [self.book])
+
+
+@unittest.skipUnless(
+    connection.vendor == "postgresql", "The current database is not PostgreSQL"
+)
+@override_settings(
+    WAGTAILSEARCH_BACKENDS={
+        "default": {
+            "BACKEND": "wagtail.search.backends.database.postgres.postgres",
+        }
+    }
+)
+class TestPostgresSearchBackendUnequalLists(TestCase):
+    """
+    Test the fix for list index out of range error when lists have different lengths.
+
+    This test addresses issue #12996, which occurred when indexing pages with empty fields,
+    especially during locale synchronization.
+    """
+
+    def setUp(self):
+        self.book_with_title_only = models.Book.objects.create(
+            title="Book with title only",
+            summary="",
+            publication_date="2025-01-01",
+            number_of_pages=100,
+        )
+
+        self.minimal_book = models.Book.objects.create(
+            title=" ", summary=None, publication_date="2025-01-01", number_of_pages=0
+        )
+
+        from wagtail.search.backends import get_search_backend
+
+        self.backend = get_search_backend()
+
+        self.assertEqual(
+            self.backend.__class__.__module__,
+            "wagtail.search.backends.database.postgres.postgres",
+        )
+
+    def test_indexing_with_empty_fields(self):
+        """Test indexing objects with empty fields doesn't cause an IndexError."""
+        try:
```

Comment:
There's no need for a try/catch here. If an `IndexError` is a test failure, then you can just leave that exception to be thrown, and it'll show up as a test failure.